v3/common/parser.go

```diff
@@ -135,6 +136,50 @@ func (p *Parser) ParseLongs(data []byte, unitCount uint32, byteOrder binary.Byte
 	return value, nil
 }
 
+// ParseFloats knows how to encode an encoded list of floats.
+func (p *Parser) ParseFloats(data []byte, unitCount uint32, byteOrder binary.ByteOrder) (value []float32, err error) {
+	defer func() {
+		if state := recover(); state != nil {
+			err = log.Wrap(state.(error))
+		}
+	}()
+
+	count := int(unitCount)
+
+	if len(data) < (TypeFloat.Size() * count) {
```

Comment:
Using this approach eliminates support in cases of data have more elements than specified in `unitCount`. Such as 

```go
// []float32{3.14159265, 2.71828182}
encoded := []byte{0x40, 0x49, 0x0f, 0xdb, 
                             0x40, 0x2d, 0xf8, 0x54}
// will raise the panic ErrNotEnoughData because
// len(encoded) != 4
value, err = p.ParseFloats(encoded, 1, TestDefaultByteOrder)
```

Reply:
That's fine. I'd rather strictness than codependency. We should only tolerate bad data if told to do so (in which case we can add options later). Otherwise, we should set our expectations exclusively from the standard.

If there's an issue in an EXIF block, this mentality helps find it rather than ignore it, and we can only choose to do that in the very beginning (it's not usually possible/preferable to make code *more strict* over time).

Reply:
Ok, I'll add the change e remove the test case where the length is above the expected